api/src/main/java/org/openmrs/module/openconceptlab/ImportServiceImpl.java

```diff
@@ -399,6 +399,15 @@ public void saveSubscription(Subscription subscription) {
 		url.setPropertyValue(prependApiIfAbsent(subscription.getUrl()));
 		adminService.saveGlobalProperty(url);
 
+		GlobalProperty apiUrl = new GlobalProperty(OpenConceptLabConstants.GP_OCL_API_URL);
+		String subscriptionUrl = adminService.getGlobalProperty(OpenConceptLabConstants.GP_SUBSCRIPTION_URL);
+		try {
+			apiUrl.setPropertyValue(new URL(subscriptionUrl).getHost());

```

Comment:
If you can always dynamically get the base url from whatever the user supplies, without hard coding it, then what is the use of the global property?